nixos/modules/services/networking/wireguard.nix

```diff
@@ -301,8 +301,10 @@ in
 
     networking.wireguard = {
 
+      enable = mkEnableOption "WireGuard";
```

Comment:
I'd add `default = cfg.interfaces != {};` here, instead of having the logic inside the `mkIf` of the implementation (and switch on `cfg.enable` there).

Thus, setting `networking.wireguard.enable` to `false` also will disable potentially configured interfaces, which might be intended anyways.

Reply:
Good idea, done!